uui/components/layout/TabList.tsx

```diff
@@ -0,0 +1,152 @@
+import { type IControlled } from '@epam/uui-core';
+import React, { forwardRef, type KeyboardEvent, type KeyboardEventHandler, ForwardedRef } from 'react';
+import { TabButton, TabButtonProps } from '../buttons/TabButton';
+import { FlexRow, FlexRowProps } from './FlexItems';
+
+type TabId = string;
+
+type TabElement = | HTMLButtonElement | HTMLAnchorElement | HTMLSpanElement;
+
+export type TabListItemProps = TabButtonProps & {
+    /** Ref to the tab button. */
+    ref?: ForwardedRef<TabElement>;
```

Comment:
By the way, using `React.ComponentProps<typeof TabButton>` would free developers from manually defining `ref`, because it would already be included. `TabButtonProps` doesn't have it.